public/app/plugins/datasource/influxdb/datasource.ts

```diff
@@ -340,17 +340,51 @@ export default class InfluxDatasource extends DataSourceWithBackend<InfluxQuery,
     let expandedQueries = queries;
     if (queries && queries.length > 0) {
       expandedQueries = queries.map((query) => {
+        if (this.isFlux) {
+          return {
+            ...query,
+            query: this.templateSrv.replace(query.query ?? '', scopedVars, 'regex'),
+          };
+        }
+
         const expandedQuery = {
           ...query,
           datasource: this.getRef(),
```

Comment:
i am guessing that the other pieces aren't important for `this.isFlux` queries, but is this change to `datasource` needed above in the flux query?

Reply:
hmm.. i have no idea, but i'd say let's put that `datasource: this.getRef()` in the flux-branch too, to be safe.

Reply:
It was there from before the migration. It appears that it's not even necessary as `...query` already has the datasource ref. Adding it in for Flux just to be safe.